CONTRIBUTING.md

```diff
@@ -0,0 +1,104 @@
+structure or algorithm designs with different complexity.
+
+**Improving comments** and **writing proper tests** are also highly welcome.
+
+### Contribution
+
+We appreciate any contribution, from fixing grammar mistakes to implementing complex algorithms. Please read this
+section if you are contributing to your work.
+
+If you submit a pull request that resolves an open issue, please help us to keep our issue list small by adding
+`fixes: #{$ISSUE_NO}` to your commit message. GitHub will use this tag to auto-close the issue if your PR is merged.
+
+#### What is an Algorithm?
+
+An Algorithm is one or more functions (or classes) that:
+
+- take one or more inputs.
+- perform some internal calculations or data manipulations.
+- return one or more outputs.
+- have minimal side effects.
+
+Algorithms should be packaged in a way that would make it easy for readers to put them into larger programs.
+
+Algorithms should:
+
+- have intuitive class and function names that make their purpose clear to readers.
+- use TypScript naming conventions and intuitive variable names to ease comprehension.
+- be flexible to take different input values.
+- raise TypeScript exceptions (RangeError, etc.) on erroneous input values.
+
+Algorithms in this repo should not be how-to examples for existing TypeScript packages. Instead, they should perform
+internal calculations or manipulations to convert input values into different output values. Those calculations or
+manipulations can use data types, classes, or functions of existing TypeScript packages but each algorithm in this repo
+should add unique value.
+
+#### File Naming Convention
+
+- filenames should use the UpperCamelCase (PascalCase) style.
+- There should be no spaces in filenames.
+- **Example:** `UserProfile.ts` is allowed but `userprofile.ts`,`Userprofile.ts`,`user-Profile.ts`,`userProfile.ts` are
+  not.
+
+#### Module System
+
+We use the [ES Module](https://hacks.mozilla.org/2018/03/es-modules-a-cartoon-deep-dive/) system, which bring an official, standardized module system to JavaScript.
+
+It roughly means you will need to use `export` and `import` statements instead of `module.exports` and `require()`.
+
+#### Coding Style
+
+To maximize the readability and correctness of our code, we require that new submissions follow the
+[TypeScript Standard Style](https://github.com/standard/ts-standard).
+
+In order to apply the coding style (where it can be done automatically). If an error is shown, please figure out what's
+wrong, fix it and run standard again.
+
+A few (but not all) of the things to keep in mind:
+
+- Use camelCase with the leading character as lowercase for identifier names (variables and functions).
+- Names start with a letter.
+- Follow code indentation: Always use 2 spaces for indentation of code blocks.
+
+```ts
+function sumOfArray(arrayOfNumbers: number[]): number {
```

Comment:
Much redundant naming: Types are made part of the name here. IMO the function should be just `sum(arr)` or `sum(nums)` or the like - the rest is clear from the type annotations.